testflight/suite_test.go

```diff
@@ -133,6 +134,25 @@ var _ = AfterEach(func() {
 	fly("destroy-pipeline", "-n", "-p", pipelineName)
 })
 
+func downloadFly(atcUrl string) (string, error) {
+	client := concourse.NewClient(atcUrl, http.DefaultClient, false)
+	readCloser, _, err := client.GetCLIReader("amd64", runtime.GOOS)
+	if err != nil {
+		return "", err
+	}
+	outFile, err := ioutil.TempFile("", "fly")
+	_, err = io.Copy(outFile, readCloser)
+	if err != nil {
+		return "", err
+	}
+	outFile.Close()
+	err = outFile.Chmod(0755)
```

Comment:
Whoops, I led you a bit astray it seems - https://ci.concourse-ci.org/builds/595287#L5e4b4571:261

Seems like we have to close the file after chmod'ing

```suggestion
	err = outFile.Chmod(0755)
	outFile.Close()
```

Reply:
it's funny, i did things in this order because macos will let you chmod open files, but linux was stricter... your suggestion is a preferable way to express the solution, I think.